src/integrationTest/java/io/aiven/kafka/connect/jdbc/ConnectRunner.java

```diff
@@ -0,0 +1,112 @@
+import org.apache.kafka.connect.runtime.Herder;
+import org.apache.kafka.connect.runtime.Worker;
+import org.apache.kafka.connect.runtime.isolation.Plugins;
+import org.apache.kafka.connect.runtime.rest.RestServer;
+import org.apache.kafka.connect.runtime.rest.entities.ConnectorInfo;
+import org.apache.kafka.connect.runtime.standalone.StandaloneConfig;
+import org.apache.kafka.connect.runtime.standalone.StandaloneHerder;
+import org.apache.kafka.connect.storage.MemoryOffsetBackingStore;
+import org.apache.kafka.connect.util.FutureCallback;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class ConnectRunner {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ConnectRunner.class);
+    private final String bootstrapServers;
+
+    private final Path pluginDir;
+
+    private Herder herder;
+
+    private Connect connect;
+
+    public ConnectRunner(final String bootstrapServers, final Path pluginDir) {
+        this.bootstrapServers = bootstrapServers;
+        this.pluginDir = pluginDir;
+    }
+
+    void start() {
+        final Map<String, String> workerProps = Map.of(
+            "bootstrap.servers", bootstrapServers,
+            "offset.flush.interval.ms", Integer.toString(5000),
+            // These don't matter much (each connector sets its own converters),
+            // but need to be filled with valid classes.
+            "key.converter", "org.apache.kafka.connect.converters.ByteArrayConverter",
+            "value.converter", "org.apache.kafka.connect.converters.ByteArrayConverter",
+            "internal.key.converter", "org.apache.kafka.connect.json.JsonConverter",
+            "internal.key.converter.schemas.enable", "false",
+            "internal.value.converter", "org.apache.kafka.connect.json.JsonConverter",
+            "internal.value.converter.schemas.enable", "false",
+            // Don't need it since we'll memory MemoryOffsetBackingStore.
+            "offset.storage.file.filename", "",
+            "plugin.path", pluginDir.toString());
+
+        final Time time = Time.SYSTEM;
+        final String workerId = "test-worker";
+        final String kafkaClusterId = "test-cluster";
+
+        final Plugins plugins = new Plugins(workerProps);
+        final StandaloneConfig config = new StandaloneConfig(workerProps);
+
+
+        final Worker worker = new Worker(workerId, time, plugins, config, new MemoryOffsetBackingStore());
+        herder = new StandaloneHerder(worker, kafkaClusterId);
+        connect = new Connect(herder, new RestServer(config));
+
+        connect.start();
+    }
+
+    public void createConnector(final Map<String, String> config) throws ExecutionException, InterruptedException {
+        assert herder != null;
+
+        final FutureCallback<Herder.Created<ConnectorInfo>> cb = new FutureCallback<>(
+            (error, info) -> {
+                if (error != null) {
+                    LOGGER.error("Failed to create job");
```

Comment:
What happens after this? I assume something wrong in a non-recoverable why. Just `throw` here?